### ISSUE TITLE
OpenSUSE container crashes host's Xorg on startup

### ISSUE THREAD

**Issue by Zialus** [jam82/buildah-molecule-images#3]
Sorry to bother you with this, but I'm not sure where else to post about this.

I was doing some tests with molecule on my machine and playing around with various docker imagines from geerlingguy, but then I realized I needed to do the tests with opensuse as the target machine, so I tried your imagine and also some of the images from [this other repo](https://github.com/mesaguy/ansible-molecule).

For some reason, what happens every time I run `molecule create`  is a crash of Xorg that closes my whole GUI session but creates the docker container successfully and leaves it running. This happens **only** with opensuse containers.

I'm not sure what could be going on, this is the [only](https://forums.docker.com/t/docker-compose-crashing-xorg-with-every-container/77940) similar post I could find.

**Comment by jomrr** [comment]
Hi Zialus, 
Thanks for supporting :)
oh oh... I think some time ago I had the same issue with my dev vm in Ubuntu ... I will have a look at it and report back.
Systemd in docker is ugly...

Greetz

**Comment by jomrr** [comment]
Could you post your molecule.yml?

**Comment by jomrr** [comment]
Have you tried it this way?

  - name: baseline-opensuse15
    image: "jam82/molecule-opensuse:15"
    override_command: false
    pre_build_image: true
    privileged: true
    tmpfs:
      - /tmp
      - /run
    volumes:
      - "/sys/fs/cgroup:/sys/fs/cgroup:ro"

**Comment by Zialus** [comment]
Same results. I've tried launching the container with pretty much all possible options in the molecule platform configuration. It always leads to the same result when using OpenSUSE based images.

I ran the container outside of the molecule script in order to get some logs, maybe this will help:

```bash
docker run -it --privileged --volume=/sys/fs/cgroup:/sys/fs/cgroup:ro jam82/molecule-opensuse
```

After crashing and logging back in:

```bash
docker ps -a                                                                                                                                                                                                                             
CONTAINER ID        IMAGE                        COMMAND                  CREATED             STATUS              PORTS               NAMES
5f09749287ac        jam82/molecule-opensuse      "/usr/lib/systemd/sy…"   6 minutes ago       Up 5 minutes                            stoic_poitras
```

```text
docker logs stoic_poitras                                                                                                                                                                                                                
systemd 234 running in system mode. (+PAM -AUDIT +SELINUX -IMA +APPARMOR -SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 -IDN default-hierarchy=hybrid)
Detected virtualization docker.
Detected architecture x86-64.

Welcome to openSUSE Leap 15.2!

Set hostname to <5f09749287ac>.
Initializing machine ID from random generator.
Couldn't move remaining userspace processes, ignoring: Input/output error
Unnecessary job for /dev/nvme0n1p6 was removed.
[  OK  ] Listening on udev Kernel Socket.
[  OK  ] Set up automount Arbitrary Executable File Formats File System Automount Point.
[  OK  ] Started Dispatch Password Requests to Console Directory Watch.
[  OK  ] Listening on udev Control Socket.
[  OK  ] Reached target Remote File Systems.
[  OK  ] Reached target Swap.
[  OK  ] Listening on /dev/initctl Compatibility Named Pipe.
[  OK  ] Created slice User and Session Slice.
[  OK  ] Listening on Journal Socket.
[  OK  ] Listening on Journal Socket (/dev/log).
[  OK  ] Reached target Local Encrypted Volumes.
[  OK  ] Created slice System Slice.
[  OK  ] Created slice system-getty.slice.
         Starting Rebuild Hardware Database...
         Starting Journal Service...
         Starting Create System Users...
         Mounting Kernel Debug File System...
[  OK  ] Reached target Slices.
         Mounting FUSE Control File System...
         Starting Apply Kernel Variables...
         Mounting Kernel Configuration File System...
         Mounting Huge Pages File System...
[  OK  ] Mounted Kernel Debug File System.
[  OK  ] Mounted Kernel Configuration File System.
[  OK  ] Mounted FUSE Control File System.
[  OK  ] Mounted Huge Pages File System.
[  OK  ] Started Create System Users.
[  OK  ] Started Apply Kernel Variables.
         Starting Create Static Device Nodes in /dev...
[  OK  ] Started Create Static Device Nodes in /dev.
         Starting udev Kernel Device Manager...
[  OK  ] Reached target Local File Systems (Pre).
[  OK  ] Reached target Local File Systems.
         Starting Rebuild Journal Catalog...
[  OK  ] Started udev Kernel Device Manager.
[  OK  ] Started Rebuild Journal Catalog.
[  OK  ] Started Journal Service.
         Starting Flush Journal to Persistent Storage...
[  OK  ] Started Flush Journal to Persistent Storage.
         Starting Create Volatile Files and Directories...
[  OK  ] Started Create Volatile Files and Directories.
         Starting Update UTMP about System Boot/Shutdown...
[  OK  ] Started Update UTMP about System Boot/Shutdown.
[  OK  ] Started Rebuild Hardware Database.
         Starting Update is Completed...
         Starting udev Coldplug all Devices...
[  OK  ] Started Update is Completed.
[  OK  ] Started udev Coldplug all Devices.
[  OK  ] Reached target System Initialization.
[  OK  ] Started Watch for changes in CA certificates.
[  OK  ] Reached target Paths.
[  OK  ] Started Daily Cleanup of Temporary Directories.
[  OK  ] Reached target Timers.
[  OK  ] Listening on D-Bus System Message Bus Socket.
[  OK  ] Reached target Sockets.
[  OK  ] Reached target Basic System.
         Starting Permit User Sessions...
[  OK  ] Started Detect if the system suffers from bsc#1089761.
[  OK  ] Started D-Bus System Message Bus.
         Starting Login Service...
         Starting Apply settings from /etc/sysconfig/keyboard...
[  OK  ] Started Permit User Sessions.
[  OK  ] Started Getty on tty1.
[  OK  ] Started Console Getty.
[  OK  ] Reached target Login Prompts.
[  OK  ] Started Login Service.
[  OK  ] Started Apply settings from /etc/sysconfig/keyboard.
[  OK  ] Reached target Multi-User System.
[  OK  ] Reached target Graphical Interface.
         Starting Update UTMP about System Runlevel Changes...
[  OK  ] Started Update UTMP about System Runlevel Changes.

Welcome to openSUSE Leap 15.2 - Kernel 5.4.80-2-MANJARO (console).


```

**Comment by jomrr** [comment]
I ran it without issues...

`docker run -it --privileged --volume=/sys/fs/cgroup:/sys/fs/cgroup:ro jam82/molecule-opensuse`

```
systemd 234 running in system mode. (+PAM -AUDIT +SELINUX -IMA +APPARMOR -SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT -GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 -IDN default-hierarchy=hybrid)
Detected virtualization docker.
Detected architecture x86-64.

Welcome to openSUSE Leap 15.2!

Set hostname to <6237fcfcf400>.
Initializing machine ID from random generator.
Couldn't move remaining userspace processes, ignoring: Input/output error
Unnecessary job for /dev/nvme0n1p2 was removed.
[  OK  ] Listening on Journal Socket.
[  OK  ] Listening on Journal Socket (/dev/log).
[  OK  ] Set up automount Arbitrary Executab…rmats File System Automount Point.
[  OK  ] Reached target Remote File Systems.
[  OK  ] Listening on udev Control Socket.
[  OK  ] Reached target Swap.
[  OK  ] Listening on /dev/initctl Compatibility Named Pipe.
[  OK  ] Started Dispatch Password Requests to Console Directory Watch.
[  OK  ] Reached target Local Encrypted Volumes.
[  OK  ] Created slice User and Session Slice.
[  OK  ] Listening on udev Kernel Socket.
[  OK  ] Created slice System Slice.
         Mounting FUSE Control File System...
         Mounting Huge Pages File System...
[  OK  ] Reached target Slices.
         Starting Create System Users...
[  OK  ] Created slice system-getty.slice.
         Starting Journal Service...
         Mounting Kernel Debug File System...
         Starting Apply Kernel Variables...
         Starting Rebuild Hardware Database...
         Mounting Kernel Configuration File System...
[  OK  ] Mounted Kernel Debug File System.
[  OK  ] Mounted Kernel Configuration File System.
[  OK  ] Mounted FUSE Control File System.
[  OK  ] Mounted Huge Pages File System.
[  OK  ] Started Create System Users.
[  OK  ] Started Apply Kernel Variables.
         Starting Create Static Device Nodes in /dev...
[  OK  ] Started Create Static Device Nodes in /dev.
         Starting udev Kernel Device Manager...
[  OK  ] Reached target Local File Systems (Pre).
[  OK  ] Reached target Local File Systems.
         Starting Rebuild Journal Catalog...
[  OK  ] Started udev Kernel Device Manager.
[  OK  ] Started Rebuild Journal Catalog.
[  OK  ] Started Journal Service.
         Starting Flush Journal to Persistent Storage...
[  OK  ] Started Flush Journal to Persistent Storage.
         Starting Create Volatile Files and Directories...
[  OK  ] Started Create Volatile Files and Directories.
         Starting Update UTMP about System Boot/Shutdown...
[  OK  ] Started Update UTMP about System Boot/Shutdown.
[  OK  ] Started Rebuild Hardware Database.
         Starting Update is Completed...
         Starting udev Coldplug all Devices...
[  OK  ] Started Update is Completed.
[  OK  ] Started udev Coldplug all Devices.
[  OK  ] Reached target System Initialization.
[  OK  ] Started Daily Cleanup of Temporary Directories.
[  OK  ] Reached target Timers.
[  OK  ] Listening on D-Bus System Message Bus Socket.
[  OK  ] Reached target Sockets.
[  OK  ] Started Watch for changes in CA certificates.
[  OK  ] Reached target Paths.
[  OK  ] Reached target Basic System.
         Starting Login Service...
[  OK  ] Started Detect if the system suffers from bsc#1089761.
         Starting Apply settings from /etc/sysconfig/keyboard...
         Starting Permit User Sessions...
[  OK  ] Started D-Bus System Message Bus.
[  OK  ] Started Apply settings from /etc/sysconfig/keyboard.
[  OK  ] Started Permit User Sessions.
[  OK  ] Started Console Getty.
[  OK  ] Started Getty on tty1.
[  OK  ] Reached target Login Prompts.
[  OK  ] Started Login Service.
[  OK  ] Reached target Multi-User System.
[  OK  ] Reached target Graphical Interface.
         Starting Update UTMP about System Runlevel Changes...
[  OK  ] Started Update UTMP about System Runlevel Changes.

Welcome to openSUSE Leap 15.2 - Kernel 5.9.11-3-MANJARO (console).


6237fcfcf400 login: 
Welcome to openSUSE Leap 15.2 - Kernel 5.9.11-3-MANJARO (console).
```

Stupid question, have you updated your Manjaro installation? I saw you are running Kernel 5.4...
I will setup an Ubuntu VM and test it... that is where I experencied it the last time.

**Comment by Zialus** [comment]
I'm using the latest LTS kernel, I guess I could try with a different kernel. I'll be trying that next and will report back later.

**Comment by Zialus** [comment]
Hi @jam82 , I have some updates.

So, as you mentioned, when using a newer kernel[see info below], Xorg doesn't crash[it still crashes when using the latest LTS]. But there's an underlying "issue" that still isn't fixed.

```
Linux manjaro 5.9.16-1-MANJARO #1 SMP PREEMPT Mon Dec 21 22:00:46 UTC 2020 x86_64 GNU/Linux
```

If I try to open a TTY with the keyboard combo (Ctrl + Alt + F1/F6), that TTY will have switched over to the OpenSUSE session!

That doesn't happen with other images, like `geerlingguy/docker-debian10-ansible` or `geerlingguy/docker-centos7-ansible` and so on.

**Comment by jomrr** [comment]
Hi Zialus,

It is because I did not remove certain systemd unit files (systemd-udevd and getty.target), therefore it spawns an agetty and attaches tty1 to a display (what it is able to do as privileged) when reaching getty.target... that is not very clean and also causes high cpu load... so I will fix it. Thanks for pointing me in the right direction.

This should also fix your problem with the latest LTS kernel, though 5.4 lacks some cgroup features...

E.g. in geerlingguys centos images he uses the official workaround from centos on docker hub.
Now I remember the reason I did not do this: That left me with a degraded systemd, which I did not really like.
But removing udevd and getty should be fine...

see: [Docker Hub CentOS](https://hub.docker.com/_/centos)
```
FROM centos:7
ENV container docker
RUN (cd /lib/systemd/system/sysinit.target.wants/; for i in *; do [ $i == \
systemd-tmpfiles-setup.service ] || rm -f $i; done); \
rm -f /lib/systemd/system/multi-user.target.wants/*;\
rm -f /etc/systemd/system/*.wants/*;\
rm -f /lib/systemd/system/local-fs.target.wants/*; \
rm -f /lib/systemd/system/sockets.target.wants/*udev*; \
rm -f /lib/systemd/system/sockets.target.wants/*initctl*; \
rm -f /lib/systemd/system/basic.target.wants/*;\
rm -f /lib/systemd/system/anaconda.target.wants/*;
VOLUME [ "/sys/fs/cgroup" ]
CMD ["/usr/sbin/init"]
```
and by the way, that cruel for loop in the first RUN command can be replaced with `find`:
`find /lib/systemd/system/sysinit.target.wants/ -type f -not -name systemd-tmpfiles-setup.service -delete`

:)
